### DISCUSSION TITLE
Wifi with tdslite

### DISCUSSION THREAD

**Discussion by redarc89** [tdslite/tdslite#9] [Q&A] [answered]
Hello,

Can we use tdslite with the WiFi library in Arduino IDE ?
I use WiFi and I need tu use tdslite but I can't une ethernet and in whole example a see you use ethernet library.

**Comment by xmkg** [comment, marked as answer]
@redarc89 of course, take a look at the tests/sketches/esp/esp.cpp as an example. The library supports all wifi/ethernet libraries compatible with the Arduino's EthernetClient interface.

**Comment by redarc89** [comment]
@mustafakemalgilor Thank you, I have another problem, can we show the result of the query request we just made in the serial monitor, the only one I see you use is affected_rows, but it not show result of query

**Reply by xmkg** [reply]
> @mustafakemalgilor Thank you, I have another problem, can we show the result of the query request we just made in the serial monitor, the only one I see you use is affected_rows, but it not show result of query

Of course, take a look into [select rows example](https://github.com/mustafakemalgilor/tdslite/blob/main/examples/arduino/03-select-rows/03-select-rows.ino#L229). The `row_callback` function will be called for each row of the result set.
